### PR TITLE
Add alternative GUID lookup fallback for description-encoded GUIDs

### LINKED ISSUE #500: On alternative GUID composition (name encoding), when device name field is shorter than available characters, remaining space is garbage memory instead of zeros
Annoying one to file an upstream bug over

Impacted mappings found in db by GUID:
```
0500000047532047616d657061640000
05000000484944204465766963650000
050000004e696d6275732b0000000000
050000004e696d6275732b008b000000
0500000047532047616d657061640000
0500000047532067616d657061640000
050000006964726f69643a636f6e0000
050000004e696d6275732b0000000000
05000000434f4d4d414e440000000000
0000000058626f782033363020576900
```

### PR DESCRIPTION
**Pull request by offalynne**
Fixed #500

### CODE REVIEW

**Comment by offalynne** [comment]
Results per current DB

```
{
    "0500000048494420":[
        [
            "05000000484944204465766963650000",
            "SteelSeries Nimbus Plus",
            "a:b0",
            "b:b1",
            "back:b15",
            "dpdown:b11",
            "dpleft:b13",
            "dpright:b12",
            "dpup:b10",
            "guide:b16",
            "leftshoulder:b4",
            "leftstick:b8",
            "lefttrigger:b6",
            "leftx:a0",
            "lefty:a1~",
            "rightshoulder:b5",
            "rightstick:b9",
            "righttrigger:b7",
            "rightx:a2",
            "righty:a3~",
            "start:b14",
            "x:b2",
            "y:b3",
            "platform:Mac OS X"
        ]
    ],
    "050000005769696d":[
        [
            "050000005769696d6f74652028303000",
            "Wii Remote",
            "a:b4",
            "b:b5",
            "back:b7",
            "dpdown:b3",
            "dpleft:b0",
            "dpright:b1",
            "dpup:b2",
            "guide:b8",
            "leftshoulder:b11",
            "lefttrigger:b12",
            "leftx:a0",
            "lefty:a1",
            "start:b6",
            "x:b10",
            "y:b9",
            "platform:Mac OS X"
        ],
        [
            "050000005769696d6f74652028313800",
            "Wii U Pro Controller",
            "a:b16",
            "b:b15",
            "back:b7",
            "dpdown:b12",
            "dpleft:b13",
            "dpright:b14",
            "dpup:b11",
            "guide:b8",
            "leftshoulder:b19",
            "leftstick:b23",
            "lefttrigger:b21",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b20",
            "rightstick:b24",
            "righttrigger:b22",
            "rightx:a2",
            "righty:a3",
            "start:b6",
            "x:b18",
            "y:b17",
            "platform:Mac OS X"
        ]
    ],
    "050000004e696d62":[
        [
            "050000004e696d6275732b0000000000",
            "SteelSeries Nimbus Plus",
            "a:b0",
            "b:b1",
            "back:b15",
            "dpdown:b11",
            "dpleft:b13",
            "dpright:b12",
            "dpup:b10",
            "guide:b16",
            "leftshoulder:b4",
            "leftstick:b8",
            "lefttrigger:b6",
            "leftx:a0",
            "lefty:a1~",
            "rightshoulder:b5",
            "rightstick:b9",
            "righttrigger:b7",
            "rightx:a2",
            "righty:a3~",
            "start:b14",
            "x:b2",
            "y:b3",
            "platform:Mac OS X"
        ],
        [
            "050000004e696d6275732b008b000000",
            "SteelSeries Nimbus Plus",
            "a:b0",
            "b:b1",
            "back:b15",
            "dpdown:b11",
            "dpleft:b13",
            "dpright:b12",
            "dpup:b10",
            "guide:b16",
            "leftshoulder:b4",
            "leftstick:b8",
            "lefttrigger:b6",
            "leftx:a0",
            "lefty:a1~",
            "rightshoulder:b5",
            "rightstick:b9",
            "righttrigger:b7",
            "rightx:a2",
            "righty:a3~",
            "start:b14",
            "x:b2",
            "y:b3",
            "platform:Mac OS X"
        ]
    ],
    "0500000047532047":[
        [
            "0500000047532047616d657061640000",
            "GameStop Gamepad",
            "a:b0",
            "b:b1",
            "back:b8",
            "dpdown:h0.4",
            "dpleft:h0.8",
            "dpright:h0.2",
            "dpup:h0.1",
            "leftshoulder:b4",
            "leftstick:b10",
            "lefttrigger:b6",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b5",
            "rightstick:b11",
            "righttrigger:b7",
            "rightx:a2",
            "righty:a3",
            "start:b9",
            "x:b2",
            "y:b3",
            "platform:Mac OS X"
        ]
    ]
}
```

```
{
    "05000000504c4159":[
        [
            "05000000504c415953544154494f4e00",
            "PS3 Controller",
            "a:b14",
            "b:b13",
            "back:b0",
            "dpdown:b6",
            "dpleft:b7",
            "dpright:b5",
            "dpup:b4",
            "guide:b16",
            "leftshoulder:b10",
            "leftstick:b1",
            "lefttrigger:b8",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b11",
            "rightstick:b2",
            "righttrigger:b9",
            "rightx:a2",
            "righty:a3",
            "start:b3",
            "x:b15",
            "y:b12",
            "platform:Linux"
        ]
    ],
    "0500000047532067":[
        [
            "0500000047532067616d657061640000",
            "GS gamepad",
            "a:b0",
            "b:b1",
            "dpdown:h0.4",
            "dpleft:h0.8",
            "dpright:h0.2",
            "dpup:h0.1",
            "guide:b8",
            "leftshoulder:b4",
            "leftstick:b10",
            "lefttrigger:b6",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b5",
            "rightstick:b11",
            "righttrigger:b7",
            "rightx:a2",
            "righty:a3",
            "start:b9",
            "x:b2",
            "y:b3",
            "platform:Linux"
        ]
    ],
    "0000000052657472":[
        [
            "00000000526574726f53746f6e653200",
            "RetroStone 2 Controller",
            "a:b1",
            "b:b0",
            "back:b10",
            "dpdown:b15",
            "dpleft:b16",
            "dpright:b17",
            "dpup:b14",
            "leftshoulder:b6",
            "lefttrigger:b8",
            "rightshoulder:b7",
            "righttrigger:b9",
            "start:b11",
            "x:b4",
            "y:b3",
            "platform:Linux"
        ]
    ],
    "xinput":[
        [
            "xinput",
            "XInput Controller",
            "a:b0",
            "b:b1",
            "back:b6",
            "dpdown:h0.4",
            "dpleft:h0.8",
            "dpright:h0.2",
            "dpup:h0.1",
            "guide:b8",
            "leftshoulder:b4",
            "leftstick:b9",
            "lefttrigger:a2",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b5",
            "rightstick:b10",
            "righttrigger:a5",
            "rightx:a3",
            "righty:a4",
            "start:b7",
            "x:b2",
            "y:b3",
            "platform:Linux"
        ]
    ],
    "050000004e696d62":[
        [
            "050000004e696d6275732b0000000000",
            "SteelSeries Nimbus Plus",
            "a:b0",
            "b:b1",
            "back:b10",
            "guide:b11",
            "leftshoulder:b4",
            "leftstick:b8",
            "lefttrigger:b6",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b5",
            "rightstick:b9",
            "righttrigger:b7",
            "rightx:a2",
            "righty:a3",
            "start:b12",
            "x:b2",
            "y:b3",
            "platform:Linux"
        ]
    ],
    "050000004c696320":[
        [
            "050000004c69632050726f20436f6e00",
            "Nintendo Switch Controller",
            "a:b0",
            "b:b1",
            "back:b8",
            "dpdown:h0.4",
            "dpleft:h0.8",
            "dpright:h0.2",
            "dpup:h0.1",
            "guide:b12",
            "leftshoulder:b4",
            "leftstick:b10",
            "lefttrigger:b6",
            "leftx:a0",
            "lefty:a1",
            "misc1:b13",
            "rightshoulder:b5",
            "rightstick:b11",
            "righttrigger:b7",
            "rightx:a2",
            "righty:a3",
            "start:b9",
            "x:b2",
            "y:b3",
            "platform:Linux"
        ]
    ],
    "050000006964726f":[
        [
            "050000006964726f69643a636f6e0000",
            "idroidcon Controller",
            "a:b1",
            "b:b2",
            "back:b8",
            "dpdown:h0.4",
            "dpleft:h0.8",
            "dpright:h0.2",
            "dpup:h0.1",
            "leftshoulder:b4",
            "leftstick:b10",
            "lefttrigger:b6",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b5",
            "rightstick:b11",
            "righttrigger:b7",
            "rightx:a2",
            "righty:a3",
            "start:b9",
            "x:b0",
            "y:b3",
            "platform:Linux"
        ]
    ],
    "050000004d4f4355":[
        [
            "050000004d4f435554452d3035335800",
            "Mocute 053X",
            "a:b0",
            "b:b1",
            "back:b8",
            "dpdown:h0.4",
            "dpleft:h0.8",
            "dpright:h0.2",
            "dpup:h0.1",
            "leftshoulder:b4",
            "leftstick:b10",
            "lefttrigger:b6",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b5",
            "rightstick:b11",
            "righttrigger:b7",
            "rightx:a2",
            "righty:a3",
            "start:b9",
            "x:b2",
            "y:b3",
            "platform:Linux"
        ],
        [
            "050000004d4f435554452d3035305800",
            "Mocute 054X",
            "a:b0",
            "b:b1",
            "back:b10",
            "dpdown:h0.4",
            "dpleft:h0.8",
            "dpright:h0.2",
            "dpup:h0.1",
            "leftshoulder:b6",
            "leftstick:b13",
            "lefttrigger:a5",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b7",
            "rightstick:b14",
            "righttrigger:a4",
            "rightx:a2",
            "righty:a3",
            "start:b11",
            "x:b3",
            "y:b4",
            "platform:Linux"
        ]
    ],
    "05000000434f4d4d":[
        [
            "05000000434f4d4d414e440000000000",
            "VX Gaming Command Series",
            "a:b0",
            "b:b1",
            "back:b8",
            "dpdown:h0.4",
            "dpleft:h0.8",
            "dpright:h0.2",
            "dpup:h0.1",
            "leftshoulder:b4",
            "leftstick:b10",
            "lefttrigger:a5",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b5",
            "rightstick:b11",
            "righttrigger:a4",
            "rightx:a2",
            "righty:a3",
            "start:b9",
            "x:b2",
            "y:b3",
            "platform:Linux"
        ]
    ],
    "0000000058626f78":[
        [
            "0000000058626f782033363020576900",
            "Xbox 360 Controller",
            "a:b0",
            "b:b1",
            "back:b14",
            "dpdown:b11",
            "dpleft:b12",
            "dpright:b13",
            "dpup:b10",
            "guide:b7",
            "leftshoulder:b4",
            "leftstick:b8",
            "lefttrigger:a2",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b5",
            "rightstick:b9",
            "righttrigger:a5",
            "rightx:a3",
            "righty:a4",
            "start:b6",
            "x:b2",
            "y:b3",
            "platform:Linux"
        ],
        [
            "0000000058626f782047616d65706100",
            "Xbox Gamepad",
            "a:b0",
            "b:b1",
            "back:b6",
            "dpdown:h0.4",
            "dpleft:h0.8",
            "dpright:h0.2",
            "dpup:h0.1",
            "guide:b8",
            "leftshoulder:b4",
            "leftstick:b9",
            "lefttrigger:a5",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b5",
            "rightstick:b10",
            "righttrigger:a4",
            "rightx:a2",
            "righty:a3",
            "start:b7",
            "x:b2",
            "y:b3",
            "platform:Linux"
        ]
    ],
    "0500000047532047":[
        [
            "0500000047532047616d657061640000",
            "GameStop Gamepad",
            "a:b0",
            "b:b1",
            "back:b8",
            "dpdown:h0.4",
            "dpleft:h0.8",
            "dpright:h0.2",
            "dpup:h0.1",
            "leftshoulder:b4",
            "leftstick:b10",
            "lefttrigger:b6",
            "leftx:a0",
            "lefty:a1",
            "rightshoulder:b5",
            "rightstick:b11",
            "righttrigger:b7",
            "rightx:a2",
            "righty:a3",
            "start:b9",
            "x:b2",
            "y:b3",
            "platform:Linux"
        ]
    ]
}
```

File: scripts/__input_system/__input_system.gml [hunk 1]
```diff
@@ -201,33 +202,50 @@ function __input_gamepad_guid_parse(_guid, _legacy, _suppress)
     }
     else
     {
-        //Check to see if this GUID fits our expected pattern:
```
Comment (offalynne) [inline scripts/__input_system/__input_system.gml:204]: type of diff GitHub's formatting makes very ugly to read sorry ¯\_(ツ)_/¯

**Comment by offalynne** [comment]
Data results per #553
[db by description linux.txt](https://github.com/JujuAdams/Input/files/10299647/db.by.description.linux.txt)
[db by description macos.txt](https://github.com/JujuAdams/Input/files/10299648/db.by.description.macos.txt)

**Comment by JujuAdams** [comment]
Thank you! I feel like we should pop `__input_gamepad_guid_parse()` out to a separate script maybe?

**Pull request merged**